### PR TITLE
Set a common group tag

### LINKED ISSUE #121: Resilience4J bulkhead metrics missing at /actutor/prometheus
**How to reproduce**
1. use two different circuit breaker (CB) instances (`cb1`, `cb2`)
2. define `resilience4j.thread-pool-bulkhead.configs.default.coreThreadPoolSiz: 5`
3. define custom config for only one CB instance `resilience4j.thread-pool-bulkhead.instances.cb2.coreThreadPoolSize: 10`
4. you will not see metrics for `cb1` (e.g. `resilience4j.bulkhead.max.thread.pool.size`) at /prometheus but /metrics is fine

**Explanation**
Prometheus requires that all meters with the same name have the same set of tag keys. See: https://github.com/micrometer-metrics/micrometer/issues/877
When we define custom config for circuit breaker, Spring Cloud Circuit Breaker (SCCB) will initialize CB instance at startup and this automatically register metric in Micrometer without any tags.
At runtime, for all CB instances, SCCB executes `Resilience4jBulkheadProvider.run()` and initializes other CB using `group` tag. See: https://github.com/spring-cloud/spring-cloud-circuitbreaker/blob/72868b72448d8bf072f168093b6bdadec609d9c7/spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JCircuitBreaker.java#L107
The result is that Prometheus will ignore metrics for `cb1`.

**Workaround**
Remove `group` tag in your code:
```java
@Bean
public MeterFilter removeGroupTagFromResilience4jBulkheadMeterFilter() {
	return new MeterFilter() {
		@Override
		public Meter.Id map(Meter.Id id) {
			if(id.getName().startsWith("resilience4j.bulkhead")) {
				List<Tag> tags = id.getTags().stream().filter(t -> !"group".equals(t.getKey())).collect(Collectors.toList());
				return id.replaceTags(tags);
			}
		   return id;
		}
	};
}
```

**Proposed fix**
Do not pass `group` tag to `Resilience4jBulkheadProvider`. Do we really need group as tag? Looks like the same value is used as name tag (created from `id`).

### PR DESCRIPTION
**Pull request by ryanjbaxter**
Fixes #121

This addresses an issue in micrometer where CB with different tags end up not showing up in the metrics.
See https://github.com/micrometer-metrics/micrometer/issues/877

### CODE REVIEW

**Comment by ryanjbaxter** [comment]
@jonatan-ivanov would like your thoughts if you have any knowledge here. 

If you think this looks OK, I would probably make this configurable as well as add a property to disable it.

File: spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JAutoConfiguration.java [hunk 1]
```diff
@@ -85,6 +86,17 @@ public Resilience4jBulkheadProvider bulkheadProvider(ThreadPoolBulkheadRegistry
 
 	}
 
+	@Configuration(proxyBeanMethods = false)
+	@ConditionalOnBean({ MeterRegistry.class })
+	public static class MicrometerResilience4JGroupCustomizerConfiguration {
+
+		@Bean
+		MeterRegistryCustomizer<MeterRegistry> metricsCommonTags() {
+			return registry -> registry.config().commonTags(Resilience4JCircuitBreaker.CIRCUIT_BREAKER_GROUP_TAG, "none");
```
Comment (jonatan-ivanov) [inline spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JAutoConfiguration.java:95]: This way the tag will be added for every single meter you have (e.g.: Spring MVC metrics, Kafka metrics, etc). Is it what you want or do you want to only add this tag to Resilience4J-related metrics?

We usually recommend adding fallback tag values at the instrumentation level, e.g.:
```
Timer timer = Timer.builder("my.timer")
    .tags("group", circuitBreaker.getGroupName() != null ? circuitBreaker.getGroupName() : "none")
```

Here's a real-life example when we attach the name of the exception as a tag but we set value "none" if there were no exceptions: [TimedAspect.java#L216-L227](https://github.com/micrometer-metrics/micrometer/blob/708c62def62c477700a22e15145be689af95fbdb/micrometer-core/src/main/java/io/micrometer/core/aop/TimedAspect.java#L216-L227). I guess your use-case is similar.

If you have no control over this, I recommend using a `MeterFilter` where you can selectively attach the missing tag based on the name and the current tags of the Meter.

Comment (ryanjbaxter) [reply spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JAutoConfiguration.java:95]: > This way the tag will be added for every single meter you have (e.g.: Spring MVC metrics, Kafka metrics, etc). Is it what you want or do you want to only add this tag to Resilience4J-related metrics?




No just resilience4j metrics, is there a better way to do that?





> If you have no control over this, I recommend using a MeterFilter where you can selectively attach the missing tag based on the name and the current tags of the Meter.

I think the problem is here https://github.com/resilience4j/resilience4j/blob/master/resilience4j-spring/src/main/java/io/github/resilience4j/bulkhead/configure/threadpool/ThreadPoolBulkheadConfiguration.java#L72-L74
Which we don't have control over.  They are not using the tags like we are.

Comment (jonatan-ivanov) [reply spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JAutoConfiguration.java:95]: I would suggest creating a `MeterFilter` (boot will auto-config it for you), like this: https://micrometer.io/docs/concepts#_transforming_metrics
So you can check if this is a resilience4j-related meter and the group tag is missing, add it with a default value.

**Review by jonatan-ivanov — APPROVED** [review]

**Pull request merged**